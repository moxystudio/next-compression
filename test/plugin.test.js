'use strict';

const zlib = require('zlib');
const withPreCompression = require('..');

const zlibBrotliCompression = zlib.brotliCompress;

const nextConfig = {
    compress: true,
};

let config = {
    plugins: [],
};

beforeEach(() => {
    config = {
        plugins: [],
    };
});

afterEach(() => {
    zlib.brotliCompress = zlibBrotliCompression;
});

it('should return the same config object that is passed', () => {
    const result = withPreCompression(nextConfig).webpack(config, {});

    expect(result).toBe(config);
});

it('should push compression settings to webpack config', () => {
    withPreCompression(nextConfig).webpack(config, {});

    expect(config).toMatchSnapshot();
});

it('should do nothing if compilation is associated to the server', () => {
    withPreCompression(nextConfig).webpack(config, { isServer: true });

    expect(config.plugins).toHaveLength(0);
});

it('should do nothing if compress is disable in the next config', () => {
    withPreCompression({ ...nextConfig, compress: false }).webpack(config, {});

    expect(config.plugins).toHaveLength(0);
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const result = withPreCompression(nextConfig).webpack(config, {});

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(nextConfig.webpack).toHaveReturnedWith(result);
});

it('should not push brotli compression plugin if brotli compression is not available', () => {
    delete zlib.brotliCompress;

    withPreCompression(nextConfig).webpack(config, {});

    expect(config).toMatchSnapshot();
});
