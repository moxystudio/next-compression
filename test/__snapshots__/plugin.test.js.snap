// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not push brotli compression plugin if brotli compression is not available 1`] = `
Object {
  "plugins": Array [
    CompressionPlugin {
      "options": Object {
        "algorithm": [Function],
        "cache": false,
        "compressionOptions": Object {
          "level": 9,
        },
        "deleteOriginalAssets": false,
        "exclude": /\\^static\\[\\\\\\\\\\\\/\\]webpack\\[\\\\\\\\\\\\/\\]/,
        "filename": "[path].gz[query]",
        "include": Array [
          /\\\\\\.atom/,
          /\\\\\\.atomcat/,
          /\\\\\\.atomsvc/,
          /\\\\\\.ccxml/,
          /\\\\\\.mpd/,
          /\\\\\\.davmount/,
          /\\\\\\.dbk/,
          /\\\\\\.xdssc/,
          /\\\\\\.ecma/,
          /\\\\\\.es/,
          /\\\\\\.emma/,
          /\\\\\\.geojson/,
          /\\\\\\.gml/,
          /\\\\\\.gpx/,
          /\\\\\\.ink/,
          /\\\\\\.inkml/,
          /\\\\\\.js/,
          /\\\\\\.mjs/,
          /\\\\\\.json/,
          /\\\\\\.map/,
          /\\\\\\.jsonml/,
          /\\\\\\.jsonld/,
          /\\\\\\.lostxml/,
          /\\\\\\.mads/,
          /\\\\\\.webmanifest/,
          /\\\\\\.mrcx/,
          /\\\\\\.mathml/,
          /\\\\\\.mscml/,
          /\\\\\\.metalink/,
          /\\\\\\.meta4/,
          /\\\\\\.mets/,
          /\\\\\\.mods/,
          /\\\\\\.opf/,
          /\\\\\\.omdoc/,
          /\\\\\\.xer/,
          /\\\\\\.pls/,
          /\\\\\\.ai/,
          /\\\\\\.eps/,
          /\\\\\\.ps/,
          /\\\\\\.pskcxml/,
          /\\\\\\.raml/,
          /\\\\\\.rdf/,
          /\\\\\\.owl/,
          /\\\\\\.rif/,
          /\\\\\\.rl/,
          /\\\\\\.rld/,
          /\\\\\\.rs/,
          /\\\\\\.rsd/,
          /\\\\\\.rss/,
          /\\\\\\.rtf/,
          /\\\\\\.sbml/,
          /\\\\\\.shf/,
          /\\\\\\.smi/,
          /\\\\\\.smil/,
          /\\\\\\.srx/,
          /\\\\\\.grxml/,
          /\\\\\\.sru/,
          /\\\\\\.ssdl/,
          /\\\\\\.ssml/,
          /\\\\\\.tei/,
          /\\\\\\.teicorpus/,
          /\\\\\\.tfi/,
          /\\\\\\.toml/,
          /\\\\\\.xdp/,
          /\\\\\\.mpkg/,
          /\\\\\\.cdxml/,
          /\\\\\\.csl/,
          /\\\\\\.wbs/,
          /\\\\\\.dart/,
          /\\\\\\.uvt/,
          /\\\\\\.uvvt/,
          /\\\\\\.es3/,
          /\\\\\\.et3/,
          /\\\\\\.kml/,
          /\\\\\\.hal/,
          /\\\\\\.zmm/,
          /\\\\\\.irp/,
          /\\\\\\.lasxml/,
          /\\\\\\.lbe/,
          /\\\\\\.xul/,
          /\\\\\\.eot/,
          /\\\\\\.dd2/,
          /\\\\\\.musicxml/,
          /\\\\\\.link66/,
          /\\\\\\.sdkm/,
          /\\\\\\.sdkd/,
          /\\\\\\.wadl/,
          /\\\\\\.xsm/,
          /\\\\\\.xdm/,
          /\\\\\\.uoml/,
          /\\\\\\.osfpvg/,
          /\\\\\\.zaz/,
          /\\\\\\.vxml/,
          /\\\\\\.wasm/,
          /\\\\\\.wsdl/,
          /\\\\\\.wspolicy/,
          /\\\\\\.ncx/,
          /\\\\\\.dtb/,
          /\\\\\\.res/,
          /\\\\\\.php/,
          /\\\\\\.pac/,
          /\\\\\\.sh/,
          /\\\\\\.tar/,
          /\\\\\\.hdd/,
          /\\\\\\.ova/,
          /\\\\\\.ovf/,
          /\\\\\\.vbox/,
          /\\\\\\.vdi/,
          /\\\\\\.vhd/,
          /\\\\\\.vmdk/,
          /\\\\\\.webapp/,
          /\\\\\\.xlf/,
          /\\\\\\.xaml/,
          /\\\\\\.xdf/,
          /\\\\\\.xenc/,
          /\\\\\\.xhtml/,
          /\\\\\\.xht/,
          /\\\\\\.xml/,
          /\\\\\\.xsl/,
          /\\\\\\.xsd/,
          /\\\\\\.rng/,
          /\\\\\\.dtd/,
          /\\\\\\.xop/,
          /\\\\\\.xpl/,
          /\\\\\\.xslt/,
          /\\\\\\.xspf/,
          /\\\\\\.mxml/,
          /\\\\\\.xhvml/,
          /\\\\\\.xvml/,
          /\\\\\\.xvm/,
          /\\\\\\.yin/,
          /\\\\\\.otf/,
          /\\\\\\.ttf/,
          /\\\\\\.bmp/,
          /\\\\\\.svg/,
          /\\\\\\.svgz/,
          /\\\\\\.psd/,
          /\\\\\\.ico/,
          /\\\\\\.bmp/,
          /\\\\\\.eml/,
          /\\\\\\.mime/,
          /\\\\\\.gltf/,
          /\\\\\\.glb/,
          /\\\\\\.dae/,
          /\\\\\\.x3d/,
          /\\\\\\.x3dz/,
          /\\\\\\.appcache/,
          /\\\\\\.manifest/,
          /\\\\\\.css/,
          /\\\\\\.csv/,
          /\\\\\\.html/,
          /\\\\\\.htm/,
          /\\\\\\.shtml/,
          /\\\\\\.jsx/,
          /\\\\\\.less/,
          /\\\\\\.markdown/,
          /\\\\\\.md/,
          /\\\\\\.mdx/,
          /\\\\\\.n3/,
          /\\\\\\.txt/,
          /\\\\\\.text/,
          /\\\\\\.conf/,
          /\\\\\\.def/,
          /\\\\\\.list/,
          /\\\\\\.log/,
          /\\\\\\.in/,
          /\\\\\\.ini/,
          /\\\\\\.rtx/,
          /\\\\\\.rtf/,
          /\\\\\\.tsv/,
          /\\\\\\.uri/,
          /\\\\\\.uris/,
          /\\\\\\.urls/,
          /\\\\\\.vcard/,
          /\\\\\\.vtt/,
          /\\\\\\.mkd/,
          /\\\\\\.org/,
          /\\\\\\.pde/,
          /\\\\\\.ymp/,
          /\\\\\\.xml/,
        ],
        "minRatio": 0.8,
        "test": /\\^static\\[\\\\\\\\\\\\/\\]/,
        "threshold": 0,
      },
    },
  ],
}
`;

exports[`should push compression settings to webpack config 1`] = `
Object {
  "plugins": Array [
    CompressionPlugin {
      "options": Object {
        "algorithm": [Function],
        "cache": false,
        "compressionOptions": Object {
          "level": 9,
        },
        "deleteOriginalAssets": false,
        "exclude": /\\^static\\[\\\\\\\\\\\\/\\]webpack\\[\\\\\\\\\\\\/\\]/,
        "filename": "[path].gz[query]",
        "include": Array [
          /\\\\\\.atom/,
          /\\\\\\.atomcat/,
          /\\\\\\.atomsvc/,
          /\\\\\\.ccxml/,
          /\\\\\\.mpd/,
          /\\\\\\.davmount/,
          /\\\\\\.dbk/,
          /\\\\\\.xdssc/,
          /\\\\\\.ecma/,
          /\\\\\\.es/,
          /\\\\\\.emma/,
          /\\\\\\.geojson/,
          /\\\\\\.gml/,
          /\\\\\\.gpx/,
          /\\\\\\.ink/,
          /\\\\\\.inkml/,
          /\\\\\\.js/,
          /\\\\\\.mjs/,
          /\\\\\\.json/,
          /\\\\\\.map/,
          /\\\\\\.jsonml/,
          /\\\\\\.jsonld/,
          /\\\\\\.lostxml/,
          /\\\\\\.mads/,
          /\\\\\\.webmanifest/,
          /\\\\\\.mrcx/,
          /\\\\\\.mathml/,
          /\\\\\\.mscml/,
          /\\\\\\.metalink/,
          /\\\\\\.meta4/,
          /\\\\\\.mets/,
          /\\\\\\.mods/,
          /\\\\\\.opf/,
          /\\\\\\.omdoc/,
          /\\\\\\.xer/,
          /\\\\\\.pls/,
          /\\\\\\.ai/,
          /\\\\\\.eps/,
          /\\\\\\.ps/,
          /\\\\\\.pskcxml/,
          /\\\\\\.raml/,
          /\\\\\\.rdf/,
          /\\\\\\.owl/,
          /\\\\\\.rif/,
          /\\\\\\.rl/,
          /\\\\\\.rld/,
          /\\\\\\.rs/,
          /\\\\\\.rsd/,
          /\\\\\\.rss/,
          /\\\\\\.rtf/,
          /\\\\\\.sbml/,
          /\\\\\\.shf/,
          /\\\\\\.smi/,
          /\\\\\\.smil/,
          /\\\\\\.srx/,
          /\\\\\\.grxml/,
          /\\\\\\.sru/,
          /\\\\\\.ssdl/,
          /\\\\\\.ssml/,
          /\\\\\\.tei/,
          /\\\\\\.teicorpus/,
          /\\\\\\.tfi/,
          /\\\\\\.toml/,
          /\\\\\\.xdp/,
          /\\\\\\.mpkg/,
          /\\\\\\.cdxml/,
          /\\\\\\.csl/,
          /\\\\\\.wbs/,
          /\\\\\\.dart/,
          /\\\\\\.uvt/,
          /\\\\\\.uvvt/,
          /\\\\\\.es3/,
          /\\\\\\.et3/,
          /\\\\\\.kml/,
          /\\\\\\.hal/,
          /\\\\\\.zmm/,
          /\\\\\\.irp/,
          /\\\\\\.lasxml/,
          /\\\\\\.lbe/,
          /\\\\\\.xul/,
          /\\\\\\.eot/,
          /\\\\\\.dd2/,
          /\\\\\\.musicxml/,
          /\\\\\\.link66/,
          /\\\\\\.sdkm/,
          /\\\\\\.sdkd/,
          /\\\\\\.wadl/,
          /\\\\\\.xsm/,
          /\\\\\\.xdm/,
          /\\\\\\.uoml/,
          /\\\\\\.osfpvg/,
          /\\\\\\.zaz/,
          /\\\\\\.vxml/,
          /\\\\\\.wasm/,
          /\\\\\\.wsdl/,
          /\\\\\\.wspolicy/,
          /\\\\\\.ncx/,
          /\\\\\\.dtb/,
          /\\\\\\.res/,
          /\\\\\\.php/,
          /\\\\\\.pac/,
          /\\\\\\.sh/,
          /\\\\\\.tar/,
          /\\\\\\.hdd/,
          /\\\\\\.ova/,
          /\\\\\\.ovf/,
          /\\\\\\.vbox/,
          /\\\\\\.vdi/,
          /\\\\\\.vhd/,
          /\\\\\\.vmdk/,
          /\\\\\\.webapp/,
          /\\\\\\.xlf/,
          /\\\\\\.xaml/,
          /\\\\\\.xdf/,
          /\\\\\\.xenc/,
          /\\\\\\.xhtml/,
          /\\\\\\.xht/,
          /\\\\\\.xml/,
          /\\\\\\.xsl/,
          /\\\\\\.xsd/,
          /\\\\\\.rng/,
          /\\\\\\.dtd/,
          /\\\\\\.xop/,
          /\\\\\\.xpl/,
          /\\\\\\.xslt/,
          /\\\\\\.xspf/,
          /\\\\\\.mxml/,
          /\\\\\\.xhvml/,
          /\\\\\\.xvml/,
          /\\\\\\.xvm/,
          /\\\\\\.yin/,
          /\\\\\\.otf/,
          /\\\\\\.ttf/,
          /\\\\\\.bmp/,
          /\\\\\\.svg/,
          /\\\\\\.svgz/,
          /\\\\\\.psd/,
          /\\\\\\.ico/,
          /\\\\\\.bmp/,
          /\\\\\\.eml/,
          /\\\\\\.mime/,
          /\\\\\\.gltf/,
          /\\\\\\.glb/,
          /\\\\\\.dae/,
          /\\\\\\.x3d/,
          /\\\\\\.x3dz/,
          /\\\\\\.appcache/,
          /\\\\\\.manifest/,
          /\\\\\\.css/,
          /\\\\\\.csv/,
          /\\\\\\.html/,
          /\\\\\\.htm/,
          /\\\\\\.shtml/,
          /\\\\\\.jsx/,
          /\\\\\\.less/,
          /\\\\\\.markdown/,
          /\\\\\\.md/,
          /\\\\\\.mdx/,
          /\\\\\\.n3/,
          /\\\\\\.txt/,
          /\\\\\\.text/,
          /\\\\\\.conf/,
          /\\\\\\.def/,
          /\\\\\\.list/,
          /\\\\\\.log/,
          /\\\\\\.in/,
          /\\\\\\.ini/,
          /\\\\\\.rtx/,
          /\\\\\\.rtf/,
          /\\\\\\.tsv/,
          /\\\\\\.uri/,
          /\\\\\\.uris/,
          /\\\\\\.urls/,
          /\\\\\\.vcard/,
          /\\\\\\.vtt/,
          /\\\\\\.mkd/,
          /\\\\\\.org/,
          /\\\\\\.pde/,
          /\\\\\\.ymp/,
          /\\\\\\.xml/,
        ],
        "minRatio": 0.8,
        "test": /\\^static\\[\\\\\\\\\\\\/\\]/,
        "threshold": 0,
      },
    },
    CompressionPlugin {
      "options": Object {
        "algorithm": [Function],
        "cache": false,
        "compressionOptions": Object {
          "level": 9,
        },
        "deleteOriginalAssets": false,
        "exclude": /\\^static\\[\\\\\\\\\\\\/\\]webpack\\[\\\\\\\\\\\\/\\]/,
        "filename": "[path].br[query]",
        "include": Array [
          /\\\\\\.atom/,
          /\\\\\\.atomcat/,
          /\\\\\\.atomsvc/,
          /\\\\\\.ccxml/,
          /\\\\\\.mpd/,
          /\\\\\\.davmount/,
          /\\\\\\.dbk/,
          /\\\\\\.xdssc/,
          /\\\\\\.ecma/,
          /\\\\\\.es/,
          /\\\\\\.emma/,
          /\\\\\\.geojson/,
          /\\\\\\.gml/,
          /\\\\\\.gpx/,
          /\\\\\\.ink/,
          /\\\\\\.inkml/,
          /\\\\\\.js/,
          /\\\\\\.mjs/,
          /\\\\\\.json/,
          /\\\\\\.map/,
          /\\\\\\.jsonml/,
          /\\\\\\.jsonld/,
          /\\\\\\.lostxml/,
          /\\\\\\.mads/,
          /\\\\\\.webmanifest/,
          /\\\\\\.mrcx/,
          /\\\\\\.mathml/,
          /\\\\\\.mscml/,
          /\\\\\\.metalink/,
          /\\\\\\.meta4/,
          /\\\\\\.mets/,
          /\\\\\\.mods/,
          /\\\\\\.opf/,
          /\\\\\\.omdoc/,
          /\\\\\\.xer/,
          /\\\\\\.pls/,
          /\\\\\\.ai/,
          /\\\\\\.eps/,
          /\\\\\\.ps/,
          /\\\\\\.pskcxml/,
          /\\\\\\.raml/,
          /\\\\\\.rdf/,
          /\\\\\\.owl/,
          /\\\\\\.rif/,
          /\\\\\\.rl/,
          /\\\\\\.rld/,
          /\\\\\\.rs/,
          /\\\\\\.rsd/,
          /\\\\\\.rss/,
          /\\\\\\.rtf/,
          /\\\\\\.sbml/,
          /\\\\\\.shf/,
          /\\\\\\.smi/,
          /\\\\\\.smil/,
          /\\\\\\.srx/,
          /\\\\\\.grxml/,
          /\\\\\\.sru/,
          /\\\\\\.ssdl/,
          /\\\\\\.ssml/,
          /\\\\\\.tei/,
          /\\\\\\.teicorpus/,
          /\\\\\\.tfi/,
          /\\\\\\.toml/,
          /\\\\\\.xdp/,
          /\\\\\\.mpkg/,
          /\\\\\\.cdxml/,
          /\\\\\\.csl/,
          /\\\\\\.wbs/,
          /\\\\\\.dart/,
          /\\\\\\.uvt/,
          /\\\\\\.uvvt/,
          /\\\\\\.es3/,
          /\\\\\\.et3/,
          /\\\\\\.kml/,
          /\\\\\\.hal/,
          /\\\\\\.zmm/,
          /\\\\\\.irp/,
          /\\\\\\.lasxml/,
          /\\\\\\.lbe/,
          /\\\\\\.xul/,
          /\\\\\\.eot/,
          /\\\\\\.dd2/,
          /\\\\\\.musicxml/,
          /\\\\\\.link66/,
          /\\\\\\.sdkm/,
          /\\\\\\.sdkd/,
          /\\\\\\.wadl/,
          /\\\\\\.xsm/,
          /\\\\\\.xdm/,
          /\\\\\\.uoml/,
          /\\\\\\.osfpvg/,
          /\\\\\\.zaz/,
          /\\\\\\.vxml/,
          /\\\\\\.wasm/,
          /\\\\\\.wsdl/,
          /\\\\\\.wspolicy/,
          /\\\\\\.ncx/,
          /\\\\\\.dtb/,
          /\\\\\\.res/,
          /\\\\\\.php/,
          /\\\\\\.pac/,
          /\\\\\\.sh/,
          /\\\\\\.tar/,
          /\\\\\\.hdd/,
          /\\\\\\.ova/,
          /\\\\\\.ovf/,
          /\\\\\\.vbox/,
          /\\\\\\.vdi/,
          /\\\\\\.vhd/,
          /\\\\\\.vmdk/,
          /\\\\\\.webapp/,
          /\\\\\\.xlf/,
          /\\\\\\.xaml/,
          /\\\\\\.xdf/,
          /\\\\\\.xenc/,
          /\\\\\\.xhtml/,
          /\\\\\\.xht/,
          /\\\\\\.xml/,
          /\\\\\\.xsl/,
          /\\\\\\.xsd/,
          /\\\\\\.rng/,
          /\\\\\\.dtd/,
          /\\\\\\.xop/,
          /\\\\\\.xpl/,
          /\\\\\\.xslt/,
          /\\\\\\.xspf/,
          /\\\\\\.mxml/,
          /\\\\\\.xhvml/,
          /\\\\\\.xvml/,
          /\\\\\\.xvm/,
          /\\\\\\.yin/,
          /\\\\\\.otf/,
          /\\\\\\.ttf/,
          /\\\\\\.bmp/,
          /\\\\\\.svg/,
          /\\\\\\.svgz/,
          /\\\\\\.psd/,
          /\\\\\\.ico/,
          /\\\\\\.bmp/,
          /\\\\\\.eml/,
          /\\\\\\.mime/,
          /\\\\\\.gltf/,
          /\\\\\\.glb/,
          /\\\\\\.dae/,
          /\\\\\\.x3d/,
          /\\\\\\.x3dz/,
          /\\\\\\.appcache/,
          /\\\\\\.manifest/,
          /\\\\\\.css/,
          /\\\\\\.csv/,
          /\\\\\\.html/,
          /\\\\\\.htm/,
          /\\\\\\.shtml/,
          /\\\\\\.jsx/,
          /\\\\\\.less/,
          /\\\\\\.markdown/,
          /\\\\\\.md/,
          /\\\\\\.mdx/,
          /\\\\\\.n3/,
          /\\\\\\.txt/,
          /\\\\\\.text/,
          /\\\\\\.conf/,
          /\\\\\\.def/,
          /\\\\\\.list/,
          /\\\\\\.log/,
          /\\\\\\.in/,
          /\\\\\\.ini/,
          /\\\\\\.rtx/,
          /\\\\\\.rtf/,
          /\\\\\\.tsv/,
          /\\\\\\.uri/,
          /\\\\\\.uris/,
          /\\\\\\.urls/,
          /\\\\\\.vcard/,
          /\\\\\\.vtt/,
          /\\\\\\.mkd/,
          /\\\\\\.org/,
          /\\\\\\.pde/,
          /\\\\\\.ymp/,
          /\\\\\\.xml/,
        ],
        "minRatio": 0.8,
        "test": /\\^static\\[\\\\\\\\\\\\/\\]/,
        "threshold": 0,
      },
    },
  ],
}
`;
